nni/retiarii/oneshot/pytorch/sampling.py

```diff
@@ -0,0 +1,316 @@
+        return sampled
+
+ENAS_REPLACE_DICT={
+    LayerChoice : PathSamplingLayerChoice,
+    InputChoice : PathSamplingInputChoice
+}
+
+class EnasModel(BaseOneShotLightningModule):
+    """
+    ENAS trainer.
+
+    Parameters
+    ----------
+    model : nn.Module
+        PyTorch model to be trained.
+    reward_function : callable
+        Receives logits and ground truth label, return a tensor, which will be feeded to RL controller as reward.
+    device : torch.device
+        ``torch.device("cpu")`` or ``torch.device("cuda")``.
+    entropy_weight : float
+        Weight of sample entropy loss.
+    skip_weight : float
+        Weight of skip penalty loss.
+    baseline_decay : float
+        Decay factor of baseline. New baseline will be equal to ``baseline_decay * baseline_old + reward * (1 - baseline_decay)``.
+    ctrl_steps_aggregate : int
+        Number of steps that will be aggregated into one mini-batch for RL controller.
+    ctrl_steps : int
+        Number of mini-batches for each epoch of RL controller learning.
+    ctrl_kwargs : dict
+        Optional kwargs that will be passed to :class:`ReinforceController`.
+    """
+
+    def __init__(self, base_model, reward_function, ctrl_kwargs = None,
+                 entropy_weight = 1e-4, skip_weight = .8, baseline_decay = .999, 
+                 ctrl_steps_aggregate = 20, grad_clip = 0, custom_replace_dict = None):
+        super().__init__(base_model, custom_replace_dict)
+
+        self.nas_fields = [ReinforceField(name, len(module),
+                                          isinstance(module, PathSamplingLayerChoice) or module.n_chosen == 1)
+                           for name, module in self.nas_modules]
+        self.controller = ReinforceController(self.nas_fields, **(ctrl_kwargs or {}))
+
+        self.reward_function = reward_function
+        
+        self.entropy_weight = entropy_weight
+        self.skip_weight = skip_weight
+        self.baseline_decay = baseline_decay
+        self.baseline = 0.
+        self.ctrl_steps_aggregate = ctrl_steps_aggregate
+        self.grad_clip = grad_clip
+
+        self.automatic_optimization = False
+    
+    def on_train_start(self) -> None:
+        train_loader = self.trainer.train_dataloader
+        val_loader = self.trainer.val_dataloaders[0]
+        self.trainer.train_dataloader = ConcatenateTrainValDataLoader(train_loader, val_loader)
+        self.trainer.val_dataloaders = None
+
+    def configure_architecture_optimizers(self):
+        return optim.Adam(self.controller.parameters(), lr=3.5e-4)
+    
+    @property
+    def default_replace_dict(self):
+        return ENAS_REPLACE_DICT
+
+    def training_step(self, batch, batch_idx):
+        # grad manually
+        opts = self.optimizers()
+        arc_opt = opts[0]
+        w_opt = opts[1:]
+
+        # optimize rnn after the model
+        batch, is_train_batch = batch
+
+        if is_train_batch:
+            # step 1 : train model params
```

Comment:
If `# step 1 :` is here, then there should be a `# step 2 : `, instead of `# step2:`